src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -3108,6 +3115,24 @@ int Pane::GetLeafPaneCount() const noexcept
     return _IsLeaf() ? 1 : (_firstChild->GetLeafPaneCount() + _secondChild->GetLeafPaneCount());
 }
 
+// Method Description:
+// - Should be called when this pane is created via a default terminal handoff
+// - Finalizes our configuration given the information that we have been
+//   created via default handoff
+void Pane::FinalizeConfigurationGivenDefault()
+{
+    if (_IsLeaf() && _profile && _profile.CloseOnExit() == CloseOnExitMode::Automatic)
+    {
+        {
+            // We only want to close 'gracefully' if we were launched by Terminal
+            // Since we were launched via defterm, override this (i.e. it will
+            // be treated as 'always', see _ControlConnectionStateChangedHandler
+            // for where this boolean is used, and see GH #13325 for discussion)
+            _isDefTermSession = true;

```

Comment:
I feel like the name of this member is a bit confusing. It doesn't just store whether this is a "def term session" right? It stores whether "is leaf pane && automatic close on exit". If that's correct, I'd give this member a different, more fitting name.

You could also write:
```cpp
_isDefTermSession = _IsLeaf() && _profile && _profile.CloseOnExit() == CloseOnExitMode::Automatic;
```
...which is IMO more robust against future changes. For instance imagine `_profile` isn't immutable and `CloseOnExit()` might return different values on each invocation. If a direct assignment is used, calling `FinalizeConfigurationGivenDefault()` (or rather `UpdateIsDefTermSession()` in that case) again would correctly update the value.

But why is this a member anyways? Can't we move this entire check to line 1020 (`if (_isDefTermSession)`)?

Reply:
Ah yeah I've fixed it so this function basically just sets `_isDefTermSession` and we check for the profile setting later when we are closing